### ISSUE TITLE
Key error `BRIGHTBLOB` in select_mock_targets

### ISSUE THREAD

**Issue by forero** [desihub/desitarget#469]
Importing `desitarget` version `0.28.0.dev2909`  and running with the following command:

```
select_mock_targets --no-spectra --nproc 4 --nside 16 --seed 10 \
    -c $SCRATCH/quicksurvey_example/targets/no_spectra/dark/input.yaml \
    --output_dir $SCRATCH/quicksurvey_example/targets/no_spectra/dark \
    --tiles $SCRATCH/quicksurvey_example/survey/subset_tiles_dark.fits
```

with the following `input.yaml`

```
# mock target configuration file
targets:
    ELG: {
        target_type: ELG,
        mockfile: '{DESI_ROOT}/mocks/DarkSky/v1.0.1/elg_0_inpt.fits',
        format: gaussianfield,
        density: 2400,
    }
    LRG: {
        target_type: LRG,
        mockfile: '{DESI_ROOT}/mocks/DarkSky/v1.0.1/lrg_0_inpt.fits',
        format: gaussianfield,
        density: 480,
    }
    QSO: {
        target_type: QSO,
        mockfile: '{DESI_ROOT}/mocks/DarkSky/v1.0.1/qso_0_inpt.fits',
        format: gaussianfield,
        zmax_qso: 1.8,
        use_simqso: True,
        density: 120,
    }
    LYA: {
        target_type: QSO,
        mockfile: '{DESI_ROOT}/mocks/lya_forest/london/v4.0/master.fits',
        format: CoLoRe,
        nside_lya: 16,
        zmin_lya: 1.8,
        density: 50,
        use_simqso: True,
        balprob: 0.0,
        add_dla: True,
    }
    MWS_MAIN: {
        target_type: STAR,
        mockfile: '{DESI_ROOT}/mocks/mws/galaxia/alpha/v0.0.6/healpix',
        nside_galaxia: 8,
        format: galaxia,
    }
    MWS_NEARBY: {
        target_type: STAR,
        mockfile: '{DESI_ROOT}/mocks/mws/100pc/v0.0.4/mock_100pc.fits',
        format: mws_100pc,
    }
    WD: {
        target_type: WD,
        mockfile: '{DESI_ROOT}/mocks/mws/wd/v1.0.0/mock_wd.fits',
        format: mws_wd,
    }
    SKY: {
        target_type: SKY,
        mockfile: '{DESI_ROOT}/mocks/uniformsky/0.2/uniformsky-2048-0.2.fits',
        format: uniformsky,
    }
```
fails with the following error:

```
Traceback (most recent call last):
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/bin/select_mock_targets", line 109, in <module>
    no_spectra=args.no_spectra)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/mock/build.py", line 884, in targets_truth
    calib_only=calib_only)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/mock/build.py", line 511, in get_spectra
    reduce=_update_spectra_status)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/internal/sharedmem.py", line 678, in map
    raise pg.get_exception()
desitarget.internal.sharedmem.SlaveException: 'BRIGHTBLOB'
Traceback (most recent call last):
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/internal/sharedmem.py", line 265, in _slaveMain
    self.main(self, *self.args)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/internal/sharedmem.py", line 563, in _main
    r = realfunc(work)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/internal/sharedmem.py", line 629, in realfunc
    else: return func(i)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/mock/build.py", line 201, in _get_spectra_onepixel
    return get_spectra_onepixel(*specargs)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/mock/build.py", line 264, in get_spectra_onepixel
    MakeMock.select_targets(chunktargets, chunktruth, targetname=data['TARGET_NAME'])
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/mock/mockmaker.py", line 3858, in select_targets
    desi_target, bgs_target, mws_target = cuts.apply_cuts(targets, tcnames=targetname)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/cuts.py", line 2074, in apply_cuts
    _prepare_optical_wise(objects, colnames=colnames)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desitarget/master/py/desitarget/cuts.py", line 1516, in _prepare_optical_wise
    brightstarinblob = (objects['BRIGHTBLOB'] & 2**0) != 0
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/conda/lib/python3.6/site-packages/astropy/table/table.py", line 1222, in __getitem__
    return self.columns[item]
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/conda/lib/python3.6/site-packages/astropy/table/table.py", line 109, in __getitem__
    return OrderedDict.__getitem__(self, item)
KeyError: 'BRIGHTBLOB'
```

Are the mocks now required to have a `BRIGHTBLOB` column? 
How should I avoid that error?

**Comment by moustakas** [comment]
The data model for DR8 changed in #466, but those changes were not propagated to `select_mock_targets`. Unfortunately my work on helping *finish* DR8 takes priority right now, so I won't be able to update the code right away...

**Comment by forero** [comment]
Thanks @moustakas @geordie666. I will try using an older tag.

I will leave this issue open as a bug in `select_mock_targets`.

**Comment by moustakas** [comment]
Fixed in #480 but please reopen if there are any other issues I didn't catch.
